File: snapshots/input/syntax/Main/QuerySyntax.cs
using System.Diagnostics.CodeAnalysis;

namespace Main;

[SuppressMessage("ReSharper", "all")]
public class QuerySyntax
{
    List<IGeneric> sourceA = new List<IGeneric>();
    List<IGeneric> sourceB = new List<IGeneric>();

    interface IGeneric
    {
        string Method();
    }

    void Select()
    {
        var x = from a in sourceA select a.Method();
    }

    void Projection()
    {
        var x = from a in sourceA select new { Name = a.Method() };
        var b = from a in x select a.Name;
    }

    void Where()
    {
        var x = from a in sourceA where a.Method().StartsWith("a") select a;
    }

    void Let()
    {
        var x = from a in sourceA
                let z = new { A = a.Method(), B = a.Method() }
                select z;
    }

    void Join()
    {
        var x = from a in sourceA
                join b in sourceB on a.Method() equals b.Method()
                select new { A = a.Method(), B = b.Method() };
    }

    void MultipleFrom()
    {
        var x = from a in sourceA
                from b in sourceB
                where a.Method() == b.Method()
                select new { A = a.Method(), B = b.Method() };
    }

    void JoinInto(List<Student> students1, List<Student> students2)
    {
        var innerGroupJoinQuery =
            from student1 in students1
            join student2 in students2 on student1.ID equals student2.ID into studentGroup
            select new { Student = student1.First, Students = studentGroup };
    }

    void Continuation(List<Student> students)
    {
        var sortedGroups =
            from student in students
            orderby student.Last, student.First
            group student by student.Last[0] into newGroup
            orderby newGroup.Key
            select newGroup;
    }

    private class Student
    {
        public string First { get; set; }
        public string Last { get; set; }
        public int ID { get; set; }
    }
}

File: snapshots/input/syntax/VBMain/QuerySyntax.vb
Imports System.Diagnostics.CodeAnalysis

Namespace VBMain
    <SuppressMessage("ReSharper", "all")>
    Public Class QuerySyntax
        Private sourceA As List(Of IGeneric) = New List(Of IGeneric)()
        Private sourceB As List(Of IGeneric) = New List(Of IGeneric)()

        Interface IGeneric
            Function Method() As String
        End Interface

        Private Sub [Select]()
            Dim x = From a In sourceA Select a.Method()
        End Sub

        Private Sub Projection()
            Dim x = From a In sourceA Select New With {Key .Name = a.Method()}
            Dim b = From a In x Select a.Name
        End Sub

        Private Sub Where()
            Dim x = From a In sourceA Where a.Method().StartsWith("a") Select a
        End Sub

        Private Sub [Let]()
            Dim x = From a In sourceA Let z = New With {Key .A = a.Method(), Key .B = a.Method()} Select z
        End Sub

        Private Sub Join()
            Dim x = From a In sourceA Join b In sourceB On a.Method() Equals b.Method() Select New With {Key .A = a.Method(), Key .B = b.Method()}
        End Sub

        Private Sub MultipleFrom()
            Dim x = From a In sourceA From b In sourceB Where a.Method() = b.Method() Select c = New With {Key .A = a.Method(), Key .B = b.Method()} Where c.A = String.Empty
        End Sub

        Private Sub Into(Students As List(Of Student))
            Dim sortedGroups = From student In Students Order By student.Last, student.First Group student By student.Last Into newGroup = Group Order By newGroup
        End Sub

        Private Class Student
            Public Property First As String
            Public Property Last As String
            Public Property ID As Integer
        End Class

    End Class
End Namespace
